addons/voice/org.openhab.voice.pollytts/src/main/java/org/openhab/voice/pollytts/internal/PollyTTSService.java

```diff
@@ -0,0 +1,240 @@
+import org.eclipse.smarthome.config.core.ConfigurableService;
+import org.eclipse.smarthome.core.audio.AudioException;
+import org.eclipse.smarthome.core.audio.AudioFormat;
+import org.eclipse.smarthome.core.audio.AudioStream;
+import org.eclipse.smarthome.core.voice.TTSException;
+import org.eclipse.smarthome.core.voice.TTSService;
+import org.eclipse.smarthome.core.voice.Voice;
+import org.openhab.voice.pollytts.internal.cloudapi.CachedPollyTTSCloudImpl;
+import org.openhab.voice.pollytts.internal.cloudapi.PollyTTSConfig;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a TTS service implementation for using Polly Text-to-Speech.
+ *
+ * @author Robert Hillman - Initial contribution
+ */
+@Component(configurationPid = SERVICE_PID, property = { Constants.SERVICE_PID + "=" + SERVICE_PID,
+        ConfigurableService.SERVICE_PROPERTY_LABEL + "=" + SERVICE_NAME + " Text-to-Speech",
+        ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=" + SERVICE_CATEGORY + ":" + SERVICE_ID,
+        ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=" + SERVICE_CATEGORY })
+public class PollyTTSService implements TTSService {
+
+    /**
+     * Service name
+     */
+    static final String SERVICE_NAME = "Polly";
+
+    /**
+     * Service id
+     */
+    static final String SERVICE_ID = "pollytts";
+
+    /**
+     * Service category
+     */
+    static final String SERVICE_CATEGORY = "voice";
+
+    /**
+     * Service pid
+     */
+    static final String SERVICE_PID = "org.openhab." + SERVICE_CATEGORY + "." + SERVICE_ID;
+
+    /**
+     * Cache folder under $userdata
+     */
+    private static final String CACHE_FOLDER_NAME = "cache";
+
+    private final Logger logger = LoggerFactory.getLogger(PollyTTSService.class);
+
+    /**
+     * We need the cached implementation to allow for FixedLengthAudioStream.
+     */
+    private CachedPollyTTSCloudImpl pollyTTSImpl;
+
+    /**
+     * Set of supported voices
+     */
+    private final Set<Voice> voices = new HashSet<>();
+
+    /**
+     * Set of supported audio formats
+     */
+    private final Set<AudioFormat> audioFormats = new HashSet<>();
+
+    private PollyTTSConfig pollyTTSConfig;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        try {
+            pollyTTSConfig = new PollyTTSConfig(config);
+            logger.debug("Using configuration {}", config);
+
+            // create cache folder
+            File userData = new File(ConfigConstants.getUserDataFolder());
+            File cacheFolder = new File(new File(userData, CACHE_FOLDER_NAME), SERVICE_PID);
```

Comment:
You do not nee a local variable for `userData`.

```suggestion
            File cacheFolder = new File(new File(new File(ConfigConstants.getUserDataFolder()), CACHE_FOLDER_NAME), SERVICE_PID);
```

Reply:
You don't even need the `new File` :smile:

```suggestion
            File cacheFolder = new File(new File(ConfigConstants.getUserDataFolder(), CACHE_FOLDER_NAME), SERVICE_PID);
```